presto-main-base/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1218,7 +1219,113 @@ public void dropMaterializedView(Session session, QualifiedObjectName viewName)
+                    materializedViews.add(new QualifiedObjectName(
+                            prefix.getCatalogName(),
+                            viewName.getSchemaName(),
+                            viewName.getTableName()));
+                }
+            }
+        }
+
+        return ImmutableList.copyOf(materializedViews);
+    }
+
+    @Override
+    public Map<QualifiedObjectName, MaterializedViewDefinition> getMaterializedViews(
+            Session session,
+            QualifiedTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, transactionManager, prefix.getCatalogName());
+        Map<QualifiedObjectName, MaterializedViewDefinition> views = new LinkedHashMap<>();
+
+        if (catalog.isPresent()) {
+            CatalogMetadata catalogMetadata = catalog.get();
+
+            for (ConnectorId connectorId : catalogMetadata.listConnectorIds()) {
+                ConnectorMetadata metadata = catalogMetadata.getMetadataFor(connectorId);
+                ConnectorSession connectorSession = session.toConnectorSession(connectorId);
+
+                List<SchemaTableName> viewNames;
+                if (prefix.getSchemaName().isPresent()) {
+                    viewNames = metadata.listMaterializedViews(connectorSession, prefix.getSchemaName().get());
+
+                    if (prefix.getTableName().isPresent()) {
+                        String tableName = prefix.getTableName().get();
+                        viewNames = viewNames.stream()
+                                .filter(name -> name.getTableName().equals(tableName))
+                                .collect(toImmutableList());
+                    }
+                }
+                else {
+                    viewNames = new ArrayList<>();
+                    for (String schemaName : metadata.listSchemaNames(connectorSession)) {
+                        viewNames.addAll(metadata.listMaterializedViews(connectorSession, schemaName));
+                    }
+                }
+
+                // Bulk retrieve definitions
+                if (!viewNames.isEmpty()) {
+                    Map<SchemaTableName, MaterializedViewDefinition> definitions;
+
+                    if (prefix.getSchemaName().isPresent()) {
+                        String schemaName = prefix.getSchemaName().get();
+                        definitions = metadata.getMaterializedViews(connectorSession, schemaName, viewNames);
+                    }
+                    else {
+                        definitions = new HashMap<>();
+                        viewNames.stream()
+                                .collect(groupingBy(SchemaTableName::getSchemaName))
+                                .forEach((schema, names) -> {
+                                    definitions.putAll(metadata.getMaterializedViews(connectorSession, schema, names));
+                                });
+                    }
```

Comment:
```suggestion
                    Map<SchemaTableName, MaterializedViewDefinition> definitions = new HashMap<>();
                    viewNames.stream()
                            .collect(groupingBy(SchemaTableName::getSchemaName))
                            .forEach((schema, names) -> {
                                definitions.putAll(metadata.getMaterializedViews(connectorSession, schema, names));
                            });
```

Could we unify these two code paths?

Taking a step further: if the `schema` parameter in `ConnectorMetadata.getMaterializedViews` could be removed, we might not even need the `groupingBy` operation here at all. What's your opinion about this?